File: app/src/main/java/com/rorpage/purtyweather/PurtyWeatherApplication.java
package com.rorpage.purtyweather;

import android.app.Application;

import com.android.volley.VolleyLog;

import timber.log.Timber;

public class PurtyWeatherApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
            VolleyLog.DEBUG = true;
        }
    }
}

File: app/src/main/java/com/rorpage/purtyweather/PurtyWeatherApplication.kt
package com.rorpage.purtyweather

import android.app.Application
import androidx.hilt.work.HiltWorkerFactory
import androidx.work.Configuration
import dagger.hilt.android.HiltAndroidApp
import javax.inject.Inject
import timber.log.Timber
import timber.log.Timber.DebugTree

@HiltAndroidApp
class PurtyWeatherApplication : Application(), Configuration.Provider {
    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        }
    }

    @Inject lateinit var workerFactory: HiltWorkerFactory

    override fun getWorkManagerConfiguration() = Configuration.Builder()
        .setWorkerFactory(workerFactory)
        .build()
}
